From 2983e58ee4796ebf24fd73ffaff90c2930cb3738 Mon Sep 17 00:00:00 2001
From: Patrick Delaunay <patrick.delaunay@foss.st.com>
Date: Thu, 12 May 2022 19:43:28 +0200
Subject: [PATCH 191/954] feat(stm32mp2): generate FIP for DDR initialization

Generate a minimal FIP used for DDR initialization for serial boot
when STM32MP_DDR_FIP_IO_STORAGE is activated.

It is loaded in internal memory before to be used with support of
the FIP memmap.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 5ba8ef9e4e27e83ca635edcdbf305dfd0d30eed8 feat(stm32mp2): generate FIP for DDR initialization]
Change-Id: I3a051ca2b258771e48c6e9fed9d77ab512c2416f
Signed-off-by: Patrick Delaunay <patrick.delaunay@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/255409
Reviewed-by: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
---
 plat/st/stm32mp2/platform.mk | 22 ++++++++++++++++++++++
 1 file changed, 22 insertions(+)

diff --git a/plat/st/stm32mp2/platform.mk b/plat/st/stm32mp2/platform.mk
index 2969732986..9a11b93c8c 100644
--- a/plat/st/stm32mp2/platform.mk
+++ b/plat/st/stm32mp2/platform.mk
@@ -236,4 +236,26 @@ check_ddr_type:
 		false; \
 	fi
 
+# generate separate DDR FIP image
+ifeq (${STM32MP_DDR_FIP_IO_STORAGE},1)
+ifneq ($(filter 1,${STM32MP_UART_PROGRAMMER} ${STM32MP_USB_PROGRAMMER}),)
+
+DDR_FIP_NAME		?=	fip-ddr.bin
+DDR_FIP_ARGS		+=	--ddr-fw ${STM32MP_DDR_FW}
+DDR_FIP_DEPS		+=	${STM32MP_DDR_FW}
+
+${BUILD_PLAT}/${DDR_FIP_NAME}: ${DDR_FIP_DEPS} fiptool
+	${Q}${FIPTOOL} create ${DDR_FIP_ARGS} $@
+	${Q}${FIPTOOL} info $@
+	@${ECHO_BLANK_LINE}
+	@echo "Built $@ successfully"
+	@${ECHO_BLANK_LINE}
+
+fip-ddr: ${BUILD_PLAT}/${DDR_FIP_NAME}
+
+fip: fip-ddr
+
+endif
+endif
+
 include plat/st/common/common_rules.mk
-- 
2.53.0

